data_structures/linked_list/singly_link_list.c

```diff
@@ -35,8 +38,27 @@ void insert()  // function to insert at first location
         start = p;
     }
 }
-///////////////////////////////////////////////////////////
-void deletion()  // function to delete from first position
+/*insert a node on index `index`*/
```

Comment:
This is an algorithm for node deletion only. We are closing this PR. Thank you for your contributions. 🙂